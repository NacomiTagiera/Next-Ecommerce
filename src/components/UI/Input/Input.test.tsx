import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";

import { Input } from ".";

describe("Input", () => {
	it("renders with default props", () => {
		const { asFragment } = render(<Input />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("assigns class names by properties", () => {
		render(<Input variant="outlined" color="error" />);

		expect(screen.getByRole("textbox")).toHaveClass(
			"rounded-md ring-1 ring-inset focus:ring-2 focus:ring-inset text-red-800 ring-red-400 hover:ring-red-500 placeholder:text-red-400 focus:ring-red-600",
		);
	});

	it("applies default class names when no props are provided", () => {
		render(<Input />);

		expect(screen.getByRole("textbox")).toHaveClass(
			"block w-full border-0 py-1.5 transition-colors duration-200 ease-in-out sm:text-sm sm:leading-6 peer focus:ring-0 text-zinc-800 ring-zinc-300 hover:ring-zinc-400 placeholder:text-zinc-400 focus:ring-skyfall-500",
		);
	});

	it("injects native input properties", () => {
		render(<Input aria-label="Type here" />);

		expect(screen.getByLabelText("Type here")).toBeVisible();
	});

	it("handles input changes", async () => {
		render(<Input />);
		const input = screen.getByRole("textbox");

		await userEvent.type(input, "Typing");
		expect(input).toHaveValue("Typing");
	});
});
