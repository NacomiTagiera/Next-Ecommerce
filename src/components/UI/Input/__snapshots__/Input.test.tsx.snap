// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with default props 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <input
      class="block w-full border-0 py-1.5 transition-colors duration-200 ease-in-out sm:text-sm sm:leading-6 peer focus:ring-0 text-zinc-800 ring-zinc-300 hover:ring-zinc-400 placeholder:text-zinc-400 focus:ring-skyfall-500"
    />
    <div
      aria-hidden="true"
      class="absolute inset-x-0 bottom-0 border-t transition-colors duration-200 ease-in-out peer-hover:border-t-2 peer-focus:border-t-2"
    />
  </div>
</DocumentFragment>
`;
